test/Container/ApplicationFactoryTest.php

```diff
@@ -1071,4 +1071,226 @@ public function testCanSpecifyRouteOptionsViaConfiguration()
         $this->assertInstanceOf(Route::class, $route);
         $this->assertEquals($expected, $route->getOptions());
     }
+
+    /**
+     * @expectedException \Zend\Expressive\Container\Exception\InvalidArgumentException
```

Comment:
> 'must be in form of an array; received "string"'

It's too generic and that substring can be reused for a different exception. (Even is recommended reuse same style of phrases for to reduce the mental effort)

`Allowed HTTP methods for a route must be in form of an array`

It's a better candidate. There is no risk of see any exception reusing the same string :)

Following my rule 2 of the less effort, just use the exact phrase because is enough short `Allowed HTTP methods for a route must be in form of an array; received "string"`